addons/medical/functions/fnc_treatment.sqf

```diff
@@ -201,6 +201,12 @@ _treatmentTime = if (isNumber (_config >> "treatmentTime")) then {
 };
 
 // Start treatment
+if (_caller != ACE_player) exitWith {
+    // It's an AI that is doing the treatment
+    sleep _treatmentTime;
+    [[_caller, _target, _selectionName, _className, _items, _usersOfItems]] call DFUNC(treatment_success);
```

Comment:
In what sense can it still fail?
From what I have seen, there would be:
alive _player (okay, I have to introduce this definitely)
"_condition" (but I have no clue about  getText (_config >> "displayNameProgress"). I guess this one is UI related and not AI.
["isnotinside"] as _exceptions in canInteractWith is needed, and I guess this is one of the things that should be checked constantly, while (alive could be checked also after waiting.

What do you think if I would introduce a new function which is only called progress, and which will be constructed close to progressBar, but instead of testing player, the caller will be tested on those conditions that aren't ui related?